pkg/endpointslice/controller_test.go

```diff
@@ -0,0 +1,197 @@
+package endpointslice
+
+import (
+	"time"
+
+	lhconstants "github.com/submariner-io/lighthouse/pkg/constants"
+	"k8s.io/apimachinery/pkg/util/wait"
+
+	"k8s.io/api/discovery/v1beta1"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+
+	"k8s.io/client-go/kubernetes"
+
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	fakeKubeClient "k8s.io/client-go/kubernetes/fake"
+	"k8s.io/client-go/rest"
+)
+
+const cluster1EndPointIP1 = "192.168.0.1"
+const cluster2EndPointIP1 = "192.169.0.1"
+const cluster1HostNamePod1 = "cl1host1"
+const cluster2HostNamePod1 = "cl2host1"
+const remoteClusterID1 = "west"
+const remoteClusterID2 = "south"
+const testName1 = "testName1"
+const testName2 = "testName2"
+const testService1 = "testService1"
+const testService2 = "testService2"
+const testNS1 = "testNameSpace1"
+const testNS2 = "testNameSpace2"
+
+var _ = Describe("EndpointSlice controller", func() {
+	t := newEndpointSliceTestDiver()
+
+	When("Endpoints exists in an EndpointSlice ", func() {
+		When("IsHealthy is called for the remote clusters", func() {
+			It("should return the appropriate response", func() {
+				esName := testName1 + remoteClusterID1
+				endPoint1 := t.newEndpoint(cluster1HostNamePod1, cluster1EndPointIP1)
+				endpointSlice := t.newEndpointSliceFromEndpoint(testService1, remoteClusterID1, esName, testNS1, []v1beta1.Endpoint{endPoint1})
+				t.createEndpointSlice(testNS1, endpointSlice)
+				t.AwaitEndpointSlice(esName, testNS1)
+				t.AwaitEpMap(testService1, testNS1)
+				Expect(t.controller.IsHealthy(KeyFunc(testService1, testNS1), remoteClusterID1)).To(BeTrue())
+			})
+		})
+	})
+
+	When("When an EndpointSlice does not exists", func() {
+		When("IsHealthy is called for the remote clusters", func() {
+			It("should return false", func() {
+				Expect(t.controller.IsHealthy(KeyFunc(testService1, testNS1), remoteClusterID1)).To(BeFalse())
+			})
+		})
+	})
+
+	When("When an EndpointSlice exists without endpoints", func() {
+		When("IsHealthy is called", func() {
+			It("should return false", func() {
+				esName := testName1 + remoteClusterID1
+				endpointSlice := t.newEndpointSliceFromEndpoint(testService1, remoteClusterID1, esName, testNS1, []v1beta1.Endpoint{})
+				t.createEndpointSlice(testNS1, endpointSlice)
+				t.AwaitEndpointSlice(esName, testNS1)
+				t.AwaitEpMap(testService1, testNS1)
+				Expect(t.controller.IsHealthy(KeyFunc(testService1, testNS1), remoteClusterID1)).To(BeFalse())
+			})
+		})
+	})
+
+	When("When multiple EndpointSlice exists with endpoints", func() {
+		When("IsHealthy is called", func() {
```

Comment:
I'm not clear on the purpose of this test - it looks like it just tests 2 separate services which is essentially covered above. Did you mean:
 
```suggestion
	When("a service exists in multiple clusters with valid endpoints", func() {
	    When("IsHealthy is called for each cluster", func() {
                 It("should return true", func() {		
```

Reply:
Done.